File: lecture03/04_largest/template.py
#!/usr/bin/env python3

import functools
import sys

# Functions

def compare_numbers(a, b):
    # TODO: Compare a and b
    return False

def main():
    sort_key = functools.cmp_to_key(compare_numbers)
    # TODO: Read numbers, sort, and print

# Main execution

if __name__ == '__main__':
    main()

File: lecture03/04_largest/template.cpp
// largest_number.cpp

#include <algorithm>
#include <iostream>
#include <sstream>
#include <string>
#include <vector>

using namespace std;

// Functions

bool read_numbers(vector<string> &numbers) {
    numbers.clear();

    string line;
    if (getline(cin, line)) {
    	stringstream ss(line);
    	string n;
    	while (ss >> n) {
    	    numbers.push_back(n);
	}
    }

    return numbers.size();
}

// Main Execution

int main(int argc, char *argv[]) {
    vector<string> numbers;

    // TODO: Read numbers, sort, and print

    return 0;
}
